src/Kovan-DssSpell.t.sol

```diff
@@ -1744,231 +1725,196 @@ contract DssSpellTest is DSTest, DSMath {
-                assertEq(xlip.wards(address(end_old)), 0);
-
-                assertEq(xlip.wards(address(esm)), 1);
-
-                OsmAbstract osm = OsmAbstract(reg.pip(ilks[i]));
-                try osm.bud(address(123)) { // Check is an OSM or Median
-                    assertEq(osm.bud(address(end)), 1);
-                    assertEq(osm.bud(address(end_old)), 0);
-                } catch {}
-            }
-        }
+        // Authorization
+        assertEq(flipYFIA.wards(address(cat))    , 0);
+        assertEq(flipYFIA.wards(address(flipMom)), 0);
 
-        // Check also old flipper for LINK-A
-        FlipAbstract oldFlipLINKA = FlipAbstract(addr.addr("MCD_FLIP_LINK_A"));
-        assertEq(oldFlipLINKA.wards(address(end)), 1);
-        assertEq(oldFlipLINKA.wards(address(end_old)), 0);
-        assertEq(oldFlipLINKA.wards(address(esm)), 1);
-
-        assertEq(vat.wards(address(end_old)), 0);
-        assertEq(cat.wards(address(end_old)), 0);
-        assertEq(vow.wards(address(end_old)), 0);
-        assertEq(pot.wards(address(end_old)), 0);
-        assertEq(spotter.wards(address(end_old)), 0);
-
-        assertEq(vat.wards(address(end)), 1);
-        assertEq(cat.wards(address(end)), 1);
-        assertEq(dog.wards(address(end)), 1);
-        assertEq(vow.wards(address(end)), 1);
-        assertEq(pot.wards(address(end)), 1);
-        assertEq(spotter.wards(address(end)), 1);
-
-        assertEq(end.wards(address(esm)), 1);
-        assertEq(vat.wards(address(esm)), 1);
-        assertEq(end_old.wards(address(esm_old)), 1);
-    }
+        assertEq(vat.wards(address(clipYFIA))    , 1);
+        assertEq(dog.wards(address(clipYFIA))    , 1);
+        assertEq(clipYFIA.wards(address(dog))    , 1);
+        assertEq(clipYFIA.wards(address(end))    , 1);
+        assertEq(clipYFIA.wards(address(clipMom)), 1);
+        assertEq(clipYFIA.wards(address(esm)), 1);
 
-    function testSpellIsCast_new_End_functionality() public {
-        vote(address(spell));
-        scheduleWaitAndCast(address(spell));
-        assertTrue(spell.done());
+        assertEq(pipYFI.bud(address(clipYFIA)), 1);
+        assertEq(pipYFI.bud(address(clipMom)), 1);
 
-        DSTokenAbstract ETH = DSTokenAbstract(addr.addr("ETH"));
-        GemJoinAbstract joinETHA = GemJoinAbstract(addr.addr("MCD_JOIN_ETH_A"));
-        FlipAbstract flipETHA = FlipAbstract(addr.addr("MCD_FLIP_ETH_A"));
+        // Force max debt ceiling for YFI-A
+        hevm.store(
+            address(vat),
+            bytes32(uint256(keccak256(abi.encode(bytes32("YFI-A"), uint256(2)))) + 3),
+            bytes32(uint256(-1))
+        );
 
-        DSTokenAbstract BAT = DSTokenAbstract(addr.addr("BAT"));
-        GemJoinAbstract joinBATA = GemJoinAbstract(addr.addr("MCD_JOIN_BAT_A"));
-        FlipAbstract flipBATA = FlipAbstract(addr.addr("MCD_FLIP_BAT_A"));
+        // Add balance to the test address
+        uint256 ilkAmt = 0.05 * 10 ** 18;
```

Comment:
I don't normally see decimals. Could you assert that `ilkAmt > 0` just so I have confidence there isn't some rounding error which puts that to 0?